x-pack/test/functional/apps/infra/hosts_view.ts

```diff
@@ -342,6 +362,17 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
               await pageObjects.assetDetails.metadataRemovePinExists();
             expect(removeFilterShouldNotExist).to.be(false);
           });
+
+          it('preserves search term between page reloads', async () => {
+            const searchInput = await pageObjects.assetDetails.getMetadataSearchField();
+
+            expect(await searchInput.getAttribute('value')).to.be('');
+
+            await searchInput.type('test');
+            await refreshPageWithDelay();
+
+            expect(await searchInput.getAttribute('value')).to.be('test');
```

Comment:
Asserts after page refresh are tricky and often causes flakiness.

 The "safest" approach would be to wrap the `expect` in a `retry.try` function.

```ts
   retry.try(async () => {
       expect(await searchInput.getAttribute('value')).to.be('test');
   })
```